packages/react/src/BlockNoteTheme.ts

```diff
@@ -85,33 +113,69 @@ export const getBlockNoteTheme = (
+          : fullTheme.colors.highlightColors.brown.text.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.red.text.dark
+          : fullTheme.colors.highlightColors.red.text.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.orange.text.dark
+          : fullTheme.colors.highlightColors.orange.text.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.yellow.text.dark
+          : fullTheme.colors.highlightColors.yellow.text.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.green.text.dark
+          : fullTheme.colors.highlightColors.green.text.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.blue.text.dark
+          : fullTheme.colors.highlightColors.blue.text.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.purple.text.dark
+          : fullTheme.colors.highlightColors.purple.text.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.pink.text.dark
+          : fullTheme.colors.highlightColors.pink.text.light,
       ],
       backgroundColors: [
-        // primaryBackground,
-        blockNoteColorScheme[0],
-        "#ebeced",
-        "#e9e5e3",
-        "#fbe4e4",
-        "#f6e9d9",
-        "#fbf3db",
-        "#ddedea",
-        "#ddebf1",
-        "#eae4f2",
-        "#f4dfeb",
+        useDarkTheme
+          ? fullTheme.colors.editor.background.dark
+          : fullTheme.colors.editor.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.gray.background.dark
+          : fullTheme.colors.highlightColors.gray.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.brown.background.dark
+          : fullTheme.colors.highlightColors.brown.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.red.background.dark
+          : fullTheme.colors.highlightColors.red.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.orange.background.dark
+          : fullTheme.colors.highlightColors.orange.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.yellow.background.dark
+          : fullTheme.colors.highlightColors.yellow.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.green.background.dark
+          : fullTheme.colors.highlightColors.green.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.blue.background.dark
+          : fullTheme.colors.highlightColors.blue.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.purple.background.dark
+          : fullTheme.colors.highlightColors.purple.background.light,
+        useDarkTheme
+          ? fullTheme.colors.highlightColors.pink.background.dark
+          : fullTheme.colors.highlightColors.pink.background.light,
       ],
     },
     components: {
```

Comment:
I'm wondering if we should make it easy to customize component styles as well for consumers.

I realize this will be mantine specific, so it should be an advanced option for sure. I also think we should only do it if it's easy to implement. The benefit I see is that it will basically allow almost full styleability of UI components (without creating your own) - even though users would probably need to dive into the source / mantine to see what exactly is stylable.

I think the main thing we need to know for this is whether we can extend from a mantine theme and keep both our `components` settings, and add new ones? i.e.: is the `components` setting extendable?

Reply:
I think it should be possible, though I don't think we should spend much time on it. You can technically already do this with just CSS, e.g.:
```css
.mantine-Menu-item {
  background-color: blue !important;
}
```
So it's only really a semantic difference for a use case that we don't expect to be too common.